agent/operator_endpoint_test.go

```diff
@@ -435,7 +435,21 @@ func TestOperator_AutopilotSetConfiguration(t *testing.T) {
 	a := NewTestAgent(t, "")
 	defer a.Shutdown()
 
+	// Provide a non-default value only for CleanupDeadServers.
+	// Expect all other fields to be updated with default values
+	// (except CreateIndex and ModifyIndex).
 	body := bytes.NewBuffer([]byte(`{"CleanupDeadServers": false}`))
+	expected := structs.AutopilotConfig{
```

Comment:
Improved test by forcing comparison of full configuration object (except CreateIndex and ModifyIndex), per review

Reply:
Side note: I got a bit tripped up between `AutopilotConfig` and `AutopilotConfiguration`